Code/BuildSystem/CMake/CMakeUtils/ezUtilsUnityFiles.cmake

```diff
@@ -63,26 +65,30 @@ function(ez_generate_folder_unity_file TARGET_NAME PROJECT_DIRECTORY SUB_FOLDER_
 		set(CUR_FILE_FULL "${PROJECT_DIRECTORY}/${SUB_FOLDER_PATH}/${CUR_FILE}")
 
 		file(TO_NATIVE_PATH ${CUR_FILE_FULL} CUR_FILE_NATIVE)
-		file(APPEND ${TMP_UNITY_FILE} "#include \"${CUR_FILE_NATIVE}\"\n")
+		string(APPEND UNITY_FILE_CONTENTS "#include \"${CUR_FILE_NATIVE}\"\n")
 
 		set_source_files_properties(${CUR_FILE_FULL} PROPERTIES HEADER_FILE_ONLY true)
 	endforeach()
+	
+	file(APPEND ${TMP_UNITY_FILE} ${UNITY_FILE_CONTENTS})
 
 	set(UNITY_FILE_NEEDS_UPDATE true)
 
 	if(EXISTS ${FOLDER_UNITY_FILE})
 		# check if the generated unity file is different from the existing one
-		file(SHA1 ${FOLDER_UNITY_FILE} UNITY_EXISTING_HASH)
-		file(SHA1 ${TMP_UNITY_FILE} UNITY_NEW_HASH)
+		#file(SHA1 ${FOLDER_UNITY_FILE} UNITY_EXISTING_HASH)
+		file(READ ${FOLDER_UNITY_FILE} OLD_UNITY_FILE_CONTENTS)
+		string(SHA1 UNITY_EXISTING_HASH ${OLD_UNITY_FILE_CONTENTS})
+		string(SHA1 UNITY_NEW_HASH ${UNITY_FILE_CONTENTS})
 
 		if(${UNITY_EXISTING_HASH} STREQUAL ${UNITY_NEW_HASH})
 			set(UNITY_FILE_NEEDS_UPDATE false)
 		endif()
 	endif()
 
 	if(${UNITY_FILE_NEEDS_UPDATE})
-		file(READ ${TMP_UNITY_FILE} UNITY_FILE_CONTENTS)
```

Comment:
Is this message really useful? I would expect it to just be spam.

Reply:
Good catch, this was just for debugging.